File: server/php/src/HostedTokenization/HostedTokenizationController.php
<?php
namespace MyApp\HostedTokenization;

use MyApp\HostedTokenization\HostedTokenizationMapper;
use MyApp\HostedTokenization\HostedTokenizationService;
use MyApp\Payment\CreatePaymentService;
use MyApp\Payment\PaymentDetailsService;
use Slim\Psr7\Request;
use Slim\Psr7\Response;
use Laminas\Diactoros\Response\JsonResponse;
use InvalidArgumentException;

class HostedTokenizationController
{

    private $mapper;
    private $hostedTokenizationService;
    private $createPaymentService;
    private $paymentDetailsService;

    public function __construct(
        HostedTokenizationMapper $mapper,
        HostedTokenizationService $hostedTokenizationService,
        CreatePaymentService $createPaymentService,
        PaymentDetailsService $paymentDetailsService
    ) {
        $this->mapper = $mapper;
        $this->hostedTokenizationService = $hostedTokenizationService;
        $this->createPaymentService = $createPaymentService;
        $this->paymentDetailsService = $paymentDetailsService;
    }

    public function getHostedTokenization(Request $request, Response $response, $args)
    {
        $createHostedTokenizationResponse = $this->hostedTokenizationService->initHostedTokenization();
        return new JsonResponse($createHostedTokenizationResponse->toObject(), 200, [], true);
    }

    public function getPaymentResponse(Request $request, Response $response, $args)
    {
        $paymentId = $request->getQueryParams()['paymentId'];

        if (empty($paymentId)) {
            throw new InvalidArgumentException("Payment id is missing!");
        }

        $paymentDetailsResponse = $this->paymentDetailsService->getPaymentDetails($paymentId);
        return new JsonResponse($paymentDetailsResponse->toObject(), 200, [], true);
    }

    public function createHostedTokenization(Request $request, Response $response, $args)
    {
        $createPaymentRequest = $this->mapper->toHostedTokenizationPaymentRequest($this->mapper->toCreateHostedTokenizationBasicDto($request));
        $createPaymentResponse = $this->createPaymentService->createPayment($createPaymentRequest);

        return new JsonResponse($createPaymentResponse->toObject(), 200, [], true);
    }
}

File: server/dotnet/HostedTokenization/HostedTokenizationController.cs
using Microsoft.AspNetCore.Mvc;
using OnlinePayments.Sdk.Domain;

namespace OnlinePayments.Sdk.Example;

[Route("api/hostedtokenization")]
[ApiController]
public class HostedTokenizationController
{
    private readonly HostedTokenizationMapper _hosteTokenizationMapper;
    private readonly HostedTokenizationService _hostedTokenizationService;
    private readonly CreatePaymentService _createPaymentService;
    private readonly PaymentDetailsService _paymentDetailsService;

    public HostedTokenizationController(
        HostedTokenizationMapper hostedTokenizationMapper,
        HostedTokenizationService hostedTokenizationService,
        CreatePaymentService createPaymentService,
        PaymentDetailsService paymentDetailsService
    )
    {
        _hosteTokenizationMapper = hostedTokenizationMapper;
        _hostedTokenizationService = hostedTokenizationService;
        _createPaymentService = createPaymentService;
        _paymentDetailsService = paymentDetailsService;
    }
    
    [HttpGet]
    public async Task<CreateHostedTokenizationResponse> GetHostedTokenization() {
        return await _hostedTokenizationService.InitHostedTokenization();
    }

    [HttpGet("outcome")]
    public async Task<PaymentDetailsResponse> GetPaymentResponse(
        [FromQuery] string paymentId
    ) {
        if (paymentId == null)
            throw new Exception("Payment id is missing!");

        return await _paymentDetailsService.GetPaymentDetails(paymentId);

    }

    [HttpPost("basic")]
    public async Task<CreatePaymentResponse> CreateHostedTokenization(
        [FromBody] CreateHostedTokenizationBasicDto createHostedTokenizationBasicDto
    ) {
        return await _createPaymentService.CreateHostedTokenizationPayment(
                _hosteTokenizationMapper.ToHostedTokenizationPaymentRequest(createHostedTokenizationBasicDto));
    }
}
